pandas/tests/scalar/period/test_period.py

```diff
@@ -632,6 +632,29 @@ def _ex(p):
         result = p.to_timestamp("5S", how="start")
         assert result == expected
 
+    def test_to_timestamp_round(self):
+        # GH 24444
+        result = Period("2020-12-31 23:59:59.9995").to_timestamp().round("1 ms")
```

Comment:
I think we want to avoid using round here and test the output of to_timestamp itself, perhaps parametrizing over different values of how (name of the test should also be changed to something related to precision). Also you'll want to explicitly construct expected using the Timestamp constructor instead of using to_timestamp again.